src/v/kafka/server/handlers/incremental_alter_configs.cc

```diff
@@ -146,7 +146,10 @@ static void parse_and_set_topic_replication_factor(
   config_resource_operation op) {
     // set property value
     if (op == config_resource_operation::set) {
-        property.value = cluster::parsing_replication_factor(value);
+        property.value = std::nullopt;
+        if (property.value) {
```

Comment:
isn't this condition now always false?

Reply:
Oops
Yes, should be `if (value)`

fixed it